extensions/notion/src/components/forms/CreatePageForm.tsx

```diff
@@ -0,0 +1,218 @@
+  }
+  function sortProperties(a: DatabaseProperty, b: DatabaseProperty) {
+    if (a.type == "title") return -1;
+    if (b.type == "title") return 1;
+    if (!databaseView?.create_properties) return 0;
+    const valueA = databaseView.create_properties.indexOf(a.id);
+    const valueB = databaseView.create_properties.indexOf(b.id);
+    if (valueA > valueB) return 1;
+    if (valueA < valueB) return -1;
+    return 0;
+  }
+
+  async function handleSubmit(values: Form.Values) {
+    const titleKey = Object.keys(values).find((key) => key.includes("property::title"));
+    if (!titleKey || !values[titleKey]) {
+      showToast({
+        style: Toast.Style.Failure,
+        title: "Title Required",
+        message: "Please set title value",
+      });
+      return;
+    }
+
+    try {
+      await showToast({ style: Toast.Style.Animated, title: "Creating page" });
+
+      if (initialDatabaseId) {
+        values.database = initialDatabaseId;
+      }
+
+      const page = await createDatabasePage(values);
+
+      if (page) {
+        await showToast({
+          style: Toast.Style.Success,
+          title: "Created page",
+          primaryAction: {
+            title: "Open Page",
+            shortcut: { modifiers: ["cmd"], key: "o" },
+            onAction: () => handleOnOpenPage(page, useRecentPages().setRecentPage),
+          },
+          secondaryAction: page.url
+            ? {
+                title: "Copy URL",
+                shortcut: { modifiers: ["cmd", "shift"], key: "c" },
+                onAction: () => {
+                  Clipboard.copy(page.url as string);
+                },
+              }
+            : undefined,
+        });
+
+        if (mutate) {
+          mutate();
+          useNavigation().pop();
+        }
+      }
+    } catch {
+      await showToast({ style: Toast.Style.Failure, title: "Failed to create page" });
+    }
+  }
+
+  function copyDeeplink(values: Form.Values) {
+    const url = "raycast://extensions/HenriChabrand/notion/create-database-page";
+    const launchContext = encodeURIComponent(JSON.stringify(values));
+    Clipboard.copy(url + "?launchContext=" + launchContext);
+    showToast({ title: "Copied deeplink to clipboard" });
+  }
+
+  if (!isLoadingDatabases && !databases.length) {
+    showToast({
+      style: Toast.Style.Failure,
+      title: "No databases found",
+      message: "Please make sure you have access to at least one database",
+    });
+  }
+
+  return (
+    <Form
+      isLoading={isLoadingDatabases || isLoadingRelationPages}
+      navigationTitle={initialDatabaseId ? "Create New Page" : "Create Database Page"}
+      actions={
+        <ActionPanel>
+          <ActionPanel.Section>
+            <Action.SubmitForm title="Create Page" icon={Icon.Plus} onSubmit={handleSubmit} />
+            <Action.SubmitForm
+              title="Copy Deeplink to Command as Configured"
+              icon={Icon.Clipboard}
+              onSubmit={copyDeeplink}
+            />
```

Comment:
Shouldn't this use Action.CreateQuicklink instead of merely copying the deeplink?

Reply:
In order to pass the form values to to `Action.CreateQuicklink`, they'd have to be managed with `useState`. That's possible, but it'd have to be an object, adding a lot of code to an already hard to read file.

In my opinion, it's not worth the clutter, but if you'd like me to do it, I can.

Reply:
Oh, I see. It's fine to leave it as-is, given how complicated it is.

Reply:
In that case, you could refactor the page to use `useForm` and use the `values` property you get from it. That would make it easier to understand, plus you could add validation pretty easily as a bonus.

Reply:
@bkeys818 waiting for your answer here in case you didn't see it 🙂

Reply:
@thomaslombart Shoot, I didn't. Thanks for the reminder! I'll get working on it.

Reply:
All right, let me know if you need some pointers on that one 😊